Style List component

rachelspencer opened the pull request:
## Description

Style the `List` page.

## Acceptance Criteria
- [x] Enlarge checkbox and trashcan for accessibility
- [x] Align all info relating to items in two rows, wrap if text gets too long for single line
- [x] Alter instructions on page for a cleaner UI
- [x] Instruct user to sign in if they aren't
- [x] Show the list owner message at the bottom of the `List` and `Manage List` page, make it sticky

## Type of Changes

Styling

## Updates

### Before
![Screenshot 2024-04-01 at 11 04 00 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/4e2ae1ea-6d36-4a66-be89-ef7836a29a29)

![Screenshot 2024-04-01 at 11 03 53 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/3cb3bc01-7f7a-4077-a217-5bca661d8a81)

![Screenshot 2024-04-01 at 11 07 39 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/79ccf663-06b9-49b2-a1bf-2f0ca7166e85)

![Screenshot 2024-04-01 at 11 07 02 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/8c3cec5a-b0c3-435e-9c27-ae8a9385b303)

![Screenshot 2024-04-01 at 11 07 29 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/6fbc082c-3e2b-4d3b-b6d2-4f8598bb985a)

![Screenshot 2024-04-01 at 11 07 17 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/f06654b9-e269-47f8-a2be-bd49934c08b1)



### After
![Screenshot 2024-04-01 at 10 50 36 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/5f6421e8-6034-4d69-93bd-2130e3caf4aa)

![Screenshot 2024-04-01 at 10 50 45 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/c9cb5cbf-d96d-4c8a-b83c-f802889e291d)

![Screenshot 2024-04-01 at 10 10 49 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/678b7b34-ea98-454a-8984-63c7ccdff68e)

![Screenshot 2024-04-01 at 10 11 05 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/142e081f-e04f-41f7-b8c4-256e78738dc6)

![Screenshot 2024-04-01 at 11 12 07 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/2fc4d2f7-8ebe-4588-b04f-c7921b879d2e)

![Screenshot 2024-04-01 at 11 11 54 PM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/111473039/2c011a56-ba37-492e-b3a5-5c339c93f022)



<!-- If UI feature, take provide screenshots -->

## Testing Steps / QA Criteria
* make sure to test instructions with responsive and desktop views
- `git pull`
- `git checkout rs-style-list-component`
- `npm ci`
- `npm start`
- open an incognito browser, make sure you're signed out and can see the message saying to sign in to use the list page
- sign in, select a list from the `Home` page that has no items in it, navigate to the `List` page and check the styling
- select a list from the `Home` page that has items in it, check the styling on the `List` page 
- Navigate to the `Manage List` page, check the list owner message is at the bottom, just above the nav bar

Note: This is all in light mode, after the light/dark mode toggle issue is merged, this issue will be tested to ensure dark mode work as it should.

rachelspencer committed:
remove duplicate radix Theme import in index.jsx

rachelspencer committed:
add responsive styling for the list owner message on the List and Manage List pages.

rachelspencer committed:
alter view height of main layout page

rachelspencer committed:
update copy on List page, make prompt a semantic italic tag

rachelspencer committed:
remove unused css file for List

rachelspencer committed:
arrange item info in 2 rows and update heading above items

rachelspencer committed:
style delete buttons

rachelspencer committed:
correct styling for checkbox prompt

rachelspencer committed:
make delete button larger, remove white background and border

rachelspencer committed:
alter delete btn css rules

rachelspencer committed:
make checkbox larger

rachelspencer committed:
alter wording for p tag that instructs user to click link to add items, and tell user to log in to use list page is no user

sillytsundere reviewed (commented):
everything looks great to me! I am eager to see it in dark mode. I also noted the css variables you commented out in the index.css file. was that go be able to work with radix when styling? 

Also, i noted that the navbar is transparent, and I can see long list's list items behind the nav bar, is this something that can be fixed? see screenshots below,
<img width="944" alt="Screenshot 2024-04-02 at 11 03 36 AM" src="https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/135680082/f5b8d878-4ae2-45e6-b132-9c2fdd939eb3">
<img width="940" alt="Screenshot 2024-04-02 at 11 03 28 AM" src="https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/135680082/69053d12-e388-4701-aef4-b7567d1e6449">

rachelspencer committed:
uncomment css variabled in index.css file

rachelspencer commented:
Hey Paige, I added back the commented out variables, it was just apart of an experiment. 

Can you pull down again, I just ran the branch (before commenting back in the css variables) and my nav bar isn't transparent. I added back in the css variables and this didn't change the nav bar either.

Let me know if the issue is still happening.

jaguilar89 commented on src/views/List.jsx:
```diff
@@ -64,13 +73,11 @@ export function List({ listPath, data, listName, userIdFromPath }) {
 			)}
-			<p>Check off items as you shop, your list will reset after 24hrs.</p>
-			<p>Only manually uncheck if you didn't make the purchase.</p>
+			<p className="checkbox-prompt">Check items will reset after 24hrs.</p>
```
Small typo here 😊.

rachelspencer replied:
Nice spot!

Thanks Jose!

jaguilar89 commented on src/components/ListOwnerMessage.jsx:
```diff
@@ -1,12 +1,17 @@
+		return <p className="owner-message">You own this list.</p>;
 	} else if (sharedListOwner) {
-		return <p>This list belongs to {sharedListOwner.ownerName}.</p>;
+		return (
+			<p className="owner-message">
+				This list belongs to {sharedListOwner.ownerName}.
+			</p>
+		);
```
Tiny nitpick, but I noticed if a list happens to not have an owner (`ownerName` is undefined), the message will be `This list belongs to .`

<img width="214" alt="image" src="https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/be28c034-b63d-4dfe-8694-ed20a691d032">

rachelspencer replied:
Thanks Jose! Not a nitpick, a good spot! Although I know why this is happening...its because there will be shared lists in your account that were created before the list owner message issue was merged. That merge added a listOwner prop, and it now needs this to display the name.

We all need to clean up our FireStore before demos for this reason :-).

jaguilar89 reviewed (approved):
This looks good! GJ.

rachelspencer committed:
correct typo in checked box instruction prompt

trushmi reviewed (approved):
Looks good to me! Great job, @rachelspencer, refactoring the List component!

rachelspencer commented:
Thank you @trushmi

polly89 commented on src/index.css:
```diff
@@ -90,3 +106,18 @@ code {
+
+.delete-icon {
+	height: 1.5em;
+}
```
I think this size looks great. I'd just doublecheck the sizing once the NavBar has been merged to make sure they have a similar weight.

rachelspencer replied:
Good idea! Noted!

polly89 commented on src/views/List.jsx:
```diff
@@ -19,12 +20,20 @@ export function List({ listPath, data, listName, userIdFromPath }) {
 				<label htmlFor="add-first-item">
-					There are no items in this list. Click this button to add your first
+					There are no items in this list. Click this link to add your first
 					items!
```
This works as expected. However, I must say that as user I'd like to see this on the landing page which is currently the `Home` page. Have you considered displaying the message there as well? Or hiding the `List` and `Manage List` page until the user sings in?

rachelspencer replied:
We can discuss this as, right not the Home page has Modals that pop up if a user interacts with it but isn't signed in.

polly89 replied:
Yes, feel free to ignore this comment. It was just an observation/non-blocking comment.  This PR meets the AC. I’d focus on addressing the merge conflict and merging the branch.

polly89 commented on src/views/List.jsx:
```diff
@@ -64,13 +73,11 @@ export function List({ listPath, data, listName, userIdFromPath }) {
-			<p>Check off items as you shop, your list will reset after 24hrs.</p>
-			<p>Only manually uncheck if you didn't make the purchase.</p>
+			<p className="checkbox-prompt">Checked items will reset after 24hrs.</p>
 			<>
```
I like the simplified wording. Good job!

polly89 commented on src/components/ListItem.jsx:
```diff
@@ -86,22 +86,33 @@ export function ListItem({ listPath, item, name }) {
+						className="delete-btn"
+						aria-label={`Delete ${name}`}
+						onClick={handleDelete}
+					>
```
This looks great! I love how easy it is on the eyes. Great job!

rachelspencer replied:
Thank you!

polly89 commented on src/views/List.jsx:
```diff
@@ -64,13 +73,11 @@ export function List({ listPath, data, listName, userIdFromPath }) {
-						<p>Item Name</p>
-						<p>Purchase Priority</p>
+						<p>Item Name & Priority</p>
 					</div>
```
`suggestion` as a user it would be really easy to associate these categories if they had similar styling to the categories themselves. You could leave the `Item Name` as is, but place `Priority` underneath it in a smaller font size and with a background color (perhaps gray or something very subtle that isn't used in the other categories).

rachelspencer replied:
Ok I'll talk to the team about this styling option!

polly89 replied:
It’s fine as is too. It’s just an idea 😊

polly89 reviewed (approved):
Great job on this ticket. It meats all the AC. I made a few non-blocking comments.

rachelspencer committed:
resolve merge conflicts from local branch rs-style-list-component to local main

rachelspencer committed:
Merge branch 'main' into rs-style-list-component
